### PR TITLE
feat: add jedis-1.4 instrumentation

### PR DESCRIPTION
**Pull request by saartochner**
Note: we include the parameters of the SET requests, but not the response from GET requests.

### CODE REVIEW

File: instrumentation/jedis/jedis-1.4/javaagent/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+muzzle {
+  pass {
+    group.set("redis.clients")
+    module.set("jedis")
+    versions.set("[1.4.0,3.0.0)")
+    assertInverse.set(true)
+  }
+}
+
+// TODO: I'm not sure what to do here
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/build.gradle:12]: Hi Ken,
I tried my best here, but I'm not sure that I did it right.
WDYT?

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisConnectionInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisConnectionInstrumentation.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisDbAttributesGetter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisDbAttributesGetter.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisInstrumentation.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisInstrumentation.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisInstrumentationModule.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisInstrumentationModule.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisNetAttributesGetter.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisNetAttributesGetter.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisRequest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisRequest.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisRequestContext.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisRequestContext.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisSingletons.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisSingletons.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisClientTest.java [hunk 10]
```diff
@@ -0,0 +1,141 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisClientTest.java:1]: RAW COPY

File: instrumentation/jedis/jedis-1.4/javaagent/build.gradle [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+muzzle {
+  pass {
+    group.set("redis.clients")
+    module.set("jedis")
+    versions.set("[1.4.0,3.0.0)")
+    assertInverse.set(true)
+  }
+}
+
+dependencies {
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/build.gradle:12]: Hi Ken,
I tried my best here, but I'm not sure that I did it right.
WDYT?

**Comment by saartochner** [comment]
I saw that the GitHub action `Version Testing / push-branch (push)` is failing in other PRs as well.
Is it mandatory? flaky?

File: instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisClientTest.java (hunk 10 above)
Comment (saartochner) [reply instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/JedisClientTest.java:1]: Added the test: `setSetTransactionCommand`

**Comment by kenfinnigan** [comment]
Usually it's only meant to fail when creating a PR. I think there are so many instrumentation versions to test, it can sometimes get flaky.

I looked at what's changed for this instrumentation, and I don't think we need to copy anything from upstream. All we need is a config customizer like we have for JDBC: https://github.com/lumigo-io/opentelemetry-java-distro/blob/main/instrumentation/jdbc/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jdbc/JdbcConfigCustomizer.java

You could copy the same class, as it's the same variable that needs to be overridden to prevent DB statement sanitization

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jedis/v1_4/JedisConfigCustomizer.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (saartochner) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jedis/v1_4/JedisConfigCustomizer.java:1]: This file was copied from `opentelemetry-java-distro/instrumentation/jdbc/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jdbc/JdbcConfigCustomizer.java`

File: instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jedis/v1_4/JedisInstrumentationModule.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.List;
+import io.opentelemetry.javaagent.instrumentation.jedis.v1_4.JedisConnectionInstrumentation;
+import io.opentelemetry.javaagent.instrumentation.jedis.v1_4.JedisInstrumentation;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumentationModule.class)
+public class JedisInstrumentationModule extends InstrumentationModule {
```
Comment (kenfinnigan) [inline instrumentation/jedis/jedis-1.4/javaagent/src/main/java/io/lumigo/javaagent/instrumentation/jedis/v1_4/JedisInstrumentationModule.java:33]: This class shouldn't be needed, as we're not overriding any instrumentation from upstream.

Our config customizer can be added without needing an instrumentation module

**Review by kenfinnigan — CHANGES_REQUESTED** [review]

File: instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/README.md [hunk 14]
```diff
@@ -0,0 +1,4 @@
+# Jedis tests
```
[the remark this answers is not in the document]
Comment (kenfinnigan) [reply instrumentation/jedis/jedis-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jedis/v1_4/README.md:1]: Sure, we can leave it in

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**